### PR TITLE
[MBL-17114][Student] - Implement  Dashboard Offline Unavailable features E2E test case

### PR DESCRIPTION
**Pull request by kdeakinstructure**
Implement  Dashboard Offline Unavailable features E2E test case
Fix some breaking offline test case (because of the resolvation of merge conflict, they were broken). Add page object methods.
Add some util functions + minor refactor on existing test cases. Attempt to fix login test 'too many attempt' problem (it's not related to offline, it's a nightly flakyness because of web login policy9.

refs: MBL-17114
affects: Student
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

File: apps/student/src/androidTest/java/com/instructure/student/ui/e2e/LoginE2ETest.kt [hunk 1]
```diff
@@ -300,6 +300,7 @@ class LoginE2ETest : StudentTest() {
 
     private fun loginWithUser(user: CanvasUserApiModel, lastSchoolSaved: Boolean = false) {
 
+        Thread.sleep(5100) //Need to wait > 5 seconds before each login attempt because of new 'too many attempts' login policy on web.
```
Comment (hermannakos) [inline apps/student/src/androidTest/java/com/instructure/student/ui/e2e/LoginE2ETest.kt:303]: We need to wait 5 seconds between the login attempts and not before each, right? We could track the login times and the time the test took to run and only wait the necessary amount of time here.

**Review by hermannakos — DISMISSED** [review]

**Review by tamaskozmer — APPROVED** [review]

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Pull request merged**